### PR TITLE
Super duper mega UPDATE

### PR DESCRIPTION
**Pull request by paolorotolo**
This PR updates all Glucosio's dependencies to latest version, including new major releases.

Breaking changes:
DATABASE - Realm:
Updated to 3.0.0 from 1.2.0 (adapted code to support new version)

CHART - MPAndroidChart:
Updated to 3.0.1 from 2.2.5.
The developer also released a tool to import data directly from Realm but after some days of testing I think the right choice would be to stick with the default library and implement a custom integration. MPAndroidChart from Realm in fact requires specific entries and it's not as flexible as the core one.

### CODE REVIEW

File: app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java [hunk 1]
```diff
@@ -389,55 +381,85 @@ private void setData() {
...
         }
 
-        chart.setData(data);
+
+        if (data.getEntryCount()!=0) {
+            chart.setData(data);
+        } else {
+            chart.setData(null);
+        }
+        Log.e("glucosio", "DATA2 is "  + data.getEntryCount());
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java]: Why log error?

File: app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java [hunk 2]
```diff
@@ -389,55 +381,85 @@ private void setData() {
...
+        chart.setNoDataTextColor(getResources().getColor(R.color.glucosio_text));
         chart.animateY(1000, Easing.EasingOption.EaseOutCubic);
         chart.invalidate();
         chart.notifyDataSetChanged();
         chart.fitScreen();
+        chart.setDescription(null);
         chart.setVisibleXRangeMaximum(20);
-        chart.moveViewToX(data.getXValCount());
+        chart.moveViewToX(data.getXMax());
+        Log.e("glucosio", "Size is: " + xValues.size());
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java]: Same

File: app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java [hunk 3]
```diff
@@ -389,55 +381,85 @@ private void setData() {
...
+
+        XAxis xAxis = chart.getXAxis();
+        xAxis.setValueFormatter(null);
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java]: Why set formatter to null and on line 421 set it back

File: app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java [hunk 4]
```diff
@@ -389,55 +381,85 @@ private void setData() {
...
+
+        final LineData finalData = data;
+        IAxisValueFormatter formatter = new IAxisValueFormatter() {
+
+            @Override
+            public String getFormattedValue(float value, AxisBase axis) {
+                Log.e("glucosio", "Values is " + value + "; Size is: " + xValues.size());
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java]: Why log error?

File: app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java [hunk 5]
```diff
@@ -389,55 +381,85 @@ private void setData() {
...
     }
 
     private LineData generateGlucoseData() {
         List<String> xVals = new ArrayList<>();
-        List<Entry> yVals = new ArrayList<>();
+        List<Entry> vals = new ArrayList<>();
 
         if (graphSpinnerRange.getSelectedItemPosition() == 0) {
             // Day view
             for (int i = 0; i < presenter.getGlucoseReadings().size(); i++) {
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java]: How about to save into variable `getGlucoseReadings` and reuse?

File: app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java [hunk 6]
```diff
@@ -389,55 +381,85 @@ private void setData() {
...
+        };
+        xAxis.setGranularity(1f); // minimum axis-step (interval) is 1
+        xAxis.setValueFormatter(formatter);
+
     }
 
     private LineData generateGlucoseData() {
         List<String> xVals = new ArrayList<>();
-        List<Entry> yVals = new ArrayList<>();
+        List<Entry> vals = new ArrayList<>();
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java]: Why such rename?

File: app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java [hunk 7]
```diff
@@ -462,93 +484,102 @@ private LineData generateGlucoseData() {
             }
         }
 
-        return new LineData(xVals,
-                generateLineDataSet(yVals, getResources().getColor(R.color.glucosio_pink)));
+        xValues = xVals;
+        LineData data = new LineData(generateLineDataSet(vals, getResources().getColor(R.color.glucosio_pink)));
+        Log.e("glucosio", "DATA1 is " + data.getXMax());
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java]: Again log error

File: app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java [hunk 8]
```diff
@@ -462,93 +484,102 @@ private LineData generateGlucoseData() {
...
     private LineData generateA1cData() {
         ArrayList<String> xVals = new ArrayList<>();
         ArrayList<Entry> yVals = new ArrayList<>();
 
-        for (int i = 0; i < presenter.getA1cReadings().size(); i++) {
+        int k = 0;
+        for (int i = presenter.getA1cReadings().size() - 1; i >= 0; i--) {
             float val = Float.parseFloat(presenter.getA1cReadings().get(i).toString());
-            yVals.add(new Entry(val, i));
+            Log.e("glucosio", "ADDING " + val);
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java]: Again

File: app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java [hunk 9]
```diff
@@ -462,93 +484,102 @@ private LineData generateGlucoseData() {
...
-            yVals.add(new Entry(val, i));
+            yVals.add(new Entry(k, val));
+            k++;
         }
 
-        xVals.clear();
-        for (int i = 0; i < presenter.getWeightReadingsDateTime().size(); i++) {
+        for (int i = presenter.getWeightReadingsDateTime().size() - 1; i >= 0; i--) {
             String date = presenter.convertDate(presenter.getWeightReadingsDateTime().get(i));
             xVals.add(date + "");
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java]: Why add empty string?

File: app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java [hunk 10]
```diff
@@ -462,93 +484,102 @@ private LineData generateGlucoseData() {
...
-            yVals.add(new Entry(val, i));
+            yVals.add(new Entry(k, val));
+            k++;
         }
 
-        xVals.clear();
-        for (int i = 0; i < presenter.getKetonesReadingsDateTime().size(); i++) {
+        for (int i = presenter.getKetonesReadingsDateTime().size() - 1; i >= 0; i--) {
             String date = presenter.convertDate(presenter.getKetonesReadingsDateTime().get(i));
             xVals.add(date + "");
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java]: Why add empty string?

File: app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java [hunk 11]
```diff
@@ -462,93 +484,102 @@ private LineData generateGlucoseData() {
...
+            Log.e("glucosio", "ADDING " + val);
+            yVals.add(new Entry(k, val));
+            k++;
         }
 
-        xVals.clear();
-        for (int i = 0; i < presenter.getA1cReadingsDateTime().size(); i++) {
+        for (int i = presenter.getA1cReadingsDateTime().size() - 1; i >= 0; i--) {
             String date = presenter.convertDate(presenter.getA1cReadingsDateTime().get(i));
             xVals.add(date + "");
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java]: Why add empty string?

File: app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java [hunk 12]
```diff
@@ -462,93 +484,102 @@ private LineData generateGlucoseData() {
...
-            yValsMin.add(new Entry(val, i));
+            yValsMin.add(new Entry(j, val));
+            j++;
         }
 
-        xVals.clear();
-        for (int i = 0; i < presenter.getPressureReadingsDateTime().size(); i++) {
+        for (int i = presenter.getPressureReadingsDateTime().size() - 1; i >= 0; i--) {
             String date = presenter.convertDate(presenter.getPressureReadingsDateTime().get(i));
             xVals.add(date + "");
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java]: Same

File: app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java [hunk 13]
```diff
@@ -558,30 +589,31 @@ private LineData generateCholesterolData() {
...
-            yVals.add(new Entry(val, i));
+            yVals.add(new Entry(k, val));
+            k++;
         }
 
-        xVals.clear();
-        for (int i = 0; i < presenter.getCholesterolReadingsDateTime().size(); i++) {
+        for (int i = presenter.getCholesterolReadingsDateTime().size() - 1; i >= 0; i--) {
             String date = presenter.convertDate(presenter.getCholesterolReadingsDateTime().get(i));
             xVals.add(date + "");
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java]: Same

File: app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java [hunk 14]
```diff
@@ -605,17 +637,13 @@ private LineDataSet generateLineDataSet(List<Entry> yVals, int color) {
...
-        set1.setDrawCubic(false);
-
         if (Build.VERSION.SDK_INT <= Build.VERSION_CODES.JELLY_BEAN_MR2) {
             set1.setDrawFilled(false);
             set1.setLineWidth(2f);
             set1.setCircleSize(4f);
             set1.setDrawCircleHole(true);
         }
-
+        Log.e("glucosio", "DATA is " + set1.getEntryCount());
```
Comment (emartynov) [inline app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java]: Same

File: app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java (hunk 1 above)
Comment (paolorotolo) [inline app/src/main/java/org/glucosio/android/fragment/OverviewFragment.java]: Forgot to remove debug logs before commit :/

**Comment by paolorotolo** [comment]
@emartynov fixed some issues, please review now.

**Pull request merged**